fishtank/src/backend/docker.ts

```diff
@@ -0,0 +1,88 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { ChildProcess, execFile, ExecFileOptions } from 'child_process'
+import { promisify } from 'util'
+
+const DEFAULT_COMMAND = 'docker'
```

Comment:
What exact docker command?

Reply:
A more important question is how are you planning to support different docker cmd. Right now container creation uses `docker run`, in the next step we will interact with node using `docker exec`. Does each `Docker` object is designed to serve one command only? If so, why? what is the pros and cons for this vs a Docker object serving more than one?

Reply:
This will be relevant to the `cmd` function type and how we know if `runDetached` is eligible to call because `docker exec` wont be able to run `runDetached` function

Reply:
> What exact docker command?

Not sure I understand the question, but this is the default command that is used to launch the Docker command line tool. In the future users may be able to customize it; for now it's used to ease testing.

> A more important question is how are you planning to support different docker cmd. Right now container creation uses docker run, in the next step we will interact with node using docker exec.

Each command will have its own method. Now there's `runDetached`, in the future there's going to be `remove`, `exec`, `createNetwork`, `createVolume` and so on.

> Does each Docker object is designed to serve one command only? If so, why? what is the pros and cons for this vs a Docker object serving more than one?

Not sure I understand this question. Once you have a `Docker` instance, you can use it to do any operation you want. With the code in this PR, you can call `runDetached` as many times as you want. It's not a "one shot" instance, if that's what you're asking.

> This will be relevant to the cmd function type and how we know if runDetached is eligible to call because docker exec wont be able to run runDetached function

Again, not sure I fully understand: `docker run --detached` and `docker exec` are two distinct commands that perform completely different operations. Why would running `docker exec` would have to "run" `runDetached`?

Reply:
I'm good on this. All the question is I am not sure if you are having `run` command in default command or in runDetached(). After you put `run` in runDetached() function, I am clear about your plan and future additional commands.